src/client.rs

```diff
@@ -114,13 +116,22 @@ impl Client {
     /// # Ok(()) }
     /// ```
     pub fn send(&self, req: impl Into<Request>) -> BoxFuture<'static, Result<Response>> {
-        let req: Request = req.into();
+        let mut req = req.into();
         let Self {
             http_client,
             middleware,
         } = self.clone();
+        let mw_stack = match req.take_middleware() {
+            Some(req_mw) => {
+                let mut mw = Vec::default();
+                mw.extend(middleware.iter().cloned());
+                mw.extend(req_mw.iter().cloned());
```

Comment:
(note to self: this one shouldn't need to be cloned since it is owned at this point.)